nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java

```diff
@@ -168,35 +170,115 @@ public boolean includeApInfoInSavedContext(AccessPath accessPath, VisitorState s
     return false;
   }
 
-  private void updateIfAssertIsPresentTrueOnOptional(
-      Context context,
+  private void handleTestAssertions(
+      VisitorState state,
+      AccessPath.AccessPathContext apContext,
+      AccessPathNullnessPropagation.Updates bothUpdates,
       MethodInvocationNode node,
+      Symbol.MethodSymbol symbol) {
+
+    Consumer<Node> nonNullMarker =
+        nonNullNode ->
+            updateNonNullAPsForOptionalContent(state.context, bothUpdates, nonNullNode, apContext);
+
+    boolean isAssertTrueMethod = methodNameUtil.isMethodAssertTrue(symbol);
+    boolean isAssertFalseMethod = methodNameUtil.isMethodAssertFalse(symbol);
+    boolean isTrueMethod = methodNameUtil.isMethodIsTrue(symbol);
+    boolean isFalseMethod = methodNameUtil.isMethodIsFalse(symbol);
+    if (isAssertTrueMethod || isAssertFalseMethod) {
+      // assertTrue(optionalFoo.isPresent())
+      // assertFalse("optional was empty", optionalFoo.isEmpty())
+      Optional<MethodInvocationNode> assertedOnMethod = getMethodForBooleanParam(node);
+      if (assertedOnMethod.isPresent()) {
+        handleBooleanAssertionOnMethod(
+            nonNullMarker,
+            state.getTypes(),
+            assertedOnMethod.get(),
+            isAssertTrueMethod,
+            isAssertFalseMethod);
+      }
+    } else if (isTrueMethod || isFalseMethod) {
+      // asertThat(optionalFoo.isPresent()).isTrue()
+      // asertThat(optionalFoo.isEmpty()).isFalse()
+      Optional<MethodInvocationNode> wrappedMethod =
+          getNodeWrappedByAssertThat(node)
+              .filter(n -> n instanceof MethodInvocationNode)
+              .map(n -> (MethodInvocationNode) n)
+              .map(this::maybeUnwrapBooleanValueOf);
+      if (wrappedMethod.isPresent()) {
+        handleBooleanAssertionOnMethod(
+            nonNullMarker, state.getTypes(), wrappedMethod.get(), isTrueMethod, isFalseMethod);
+      }
+    } else if (methodNameUtil.isMethodThatEnsuresOptionalPresent(symbol)) {
+      // assertThat(optionalRef).isPresent()
+      // assertThat(methodReturningOptional()).isNotEmpty()
+      // assertThat(mapWithOptionalValues.get("key")).isNotEmpty()
+      getNodeWrappedByAssertThat(node).ifPresent(nonNullMarker);
+    }
+  }
+
+  private Optional<MethodInvocationNode> getMethodForBooleanParam(MethodInvocationNode node) {
```

Comment:
Wonder if we can name this something like `getJUnitMethodForBooleanParam` and provide a quick javadoc. I get that this method is private and also that it will technically work on any assertion method which takes a single boolean, _but_ I think having some clue about its usage when looking at this method in isolation, beyond the comment right below, would be useful.

Reply:
i wasnt happy with the method name but i thought it should prefer to be named after what it does, not after what it is used for...
looking at it again, the method body can be simplified, so i've decided to inline this logic again in [5f95a6f](https://github.com/uber/NullAway/pull/718/commits/5f95a6f6429ecba5dfe7c0a11d5384e3eb70e410)

let me know if you want me to split it out again with the suggested name

Reply:
If we want to use what the method does strictly, then maybe `getFirstBooleanParamAsMethodInvocation` or something similar. That said, I agree the fix in [5f95a6f](https://github.com/uber/NullAway/pull/718/commits/5f95a6f6429ecba5dfe7c0a11d5384e3eb70e410) is cleaner! 🙂